src/sentry/static/sentry/app/views/admin/adminSettings.tsx

```diff
@@ -20,23 +20,35 @@ const optionsAvailable = [
   'beacon.anonymous',
 ];
 
-export default class AdminSettings extends AsyncView {
-  getEndpoints() {
-    return [['data', this.getEndpoint()]];
-  }
+type Field = ReturnType<typeof getOption>;
+
+type FieldDef = {
+  field: Field;
+  value: string | undefined;
+};
+
+type State = AsyncView['state'] & {
+  data: Record<string, FieldDef>;
```

Comment:
```suggestion
  data: Record<string, FieldDef> | null;
```

Reply:
AsyncView ensure's it won't be null once it's renderable

Reply:
that is true and maybe we should discuss it again in the next tsc, but for now, I remember that we decided to always set null in this case

Reply:
Ah I hadn't realized that we discussed this. Do you happen to know when / have a link to the meeting notes? It might have been one I missed.

Reply:
Having the null type helps reflect what can actually happen, as AsyncComponent sets all the state values to `null` when requests are made.

Reply:
Right. I think I might be looking at this from more of a practical sense. Since were only using data here in the renderBody, it must be set